version: "3.1" 
services:
  nginx:
    #restart: always
    image: pets_nginx:1.0
    #container_name: nginx_https

    deploy:
      resources:
        limits:
          #cpus: '0.50'
          memory: 500M
        
      #placement:
      #  constraints: [node.hostname == pets-mvp]     

    ports:
     - "443:443"
     #../../../../pets_hadoop/pets_v1/sourceCode/hadoop/cert/
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - ./nginx.conf:/etc/nginx/conf.d/nginx.conf 
      - ./cert/moda_server_no_pw.key:/etc/ssl/certs/server.key 
      - ./cert/moda_cert_chain.pem:/etc/ssl/certs/server.crt 
      - ./cert/eCA1_GTLSCA.crt:/etc/ssl/certs/ca.crt
    
    environment:
    #  - PWD123:/run/secrets/my-pw2
      - TZ=Asia/Taipei

    #entrypoint: redis-server
    #command:   --requirepass 'citcw200'

    networks:
      mynetwork:
        #ipv4_address: 172.26.1.88
        aliases: 
          - nginx_compose  


  web:
    #restart: always
    image: pets_flask:1.0 #flask-redis_citc:05.1
    secrets:
      - maria_file
      - hadoop_file
    ports:
     - "5916:5088"
#    - "3321:22"
    volumes:
      - ./APP__:/app/app/devp
      - ./config.py:/app/config.py
      - /etc/localtime:/etc/localtime:ro
      - ../hadoop/dataConfig:/home/hadoop/proj_/dataConfig
      - ../hadoop/dataMac:/home/hadoop/proj_/dataMac
      - ../hadoop/data_check:/home/hadoop/proj_/data_check
      - ../hadoop/data_error:/home/hadoop/proj_/data_error

    entrypoint: gunicorn
    #command:  --log-level=debug --preload -w 1 -b 0.0.0.0:5088 --timeout 600 app:app
    command:  --certfile /app/app/devp/cert.pem --keyfile /app/app/devp/key.pem  --ssl-version 2 --ciphers ECDHE-RSA-AES256-GCM-SHA384 --log-level=debug --preload -w 1 -b 0.0.0.0:5088 --timeout 600 app:app
    environment:
      - REDIS_HOST:redis_dir
      - REDIS_PORT:6387
      #- REDIS_PWD:/run/secrets/my-pw2
      - REDIS_PWD:"citcw200" 
  
    networks:
      mynetwork:
        ipv4_address: 172.28.1.3
        aliases: 
          - flask_compose 
    depends_on:
      - redis 

  OpenApi_web:
    #restart: always
    image: pets_flask:1.0 #flask-redis_citc:05.1
    secrets:
      - maria_file


    ports:
     - "5988:5088"
       # - "18088:5088"
#     - "3341:22"
    volumes:
      - ./APP__:/app/app/devp
      - ./config.py:/app/config.py
      - /etc/localtime:/etc/localtime:ro
      - ../hadoop/dataConfig:/home/hadoop/proj_/dataConfig
        #- /home/myuser/jc_swagger_ui/flasgger:/usr/local/lib/python2.7/dist-packages/flasgger
      - ../hadoop/dataMac:/home/hadoop/proj_/dataMac
      - ../hadoop/data_check:/home/hadoop/proj_/data_check
      - ../hadoop/data_error:/home/hadoop/proj_/data_error  
      - /etc/hostname:/host_name

    entrypoint: gunicorn
    #command:  --certfile /app/app/devp/cert.pem --keyfile /app/app/devp/key.pem  --ssl-version 2 --ciphers AES256-GCM-SHA384 --log-level=debug --preload -w 1 -b 0.0.0.0:5088 --timeout 600 app:app
    command:  --certfile /app/app/devp/cert.pem --keyfile /app/app/devp/key.pem  --ssl-version 2 --ciphers ECDHE-RSA-AES256-GCM-SHA384 --log-level=debug --preload -w 1 -b 0.0.0.0:5088 --timeout 600 app:app
    #gunicorn --certfile cert.pem --keyfile key.pem -b 0.0.0.0:8000 hello:app
    environment:
      - REDIS_HOST:redis_dir
      - REDIS_PORT:6379
      - REDIS_PWD:"citcw200"
    networks:
      mynetwork:
        ipv4_address: 172.27.1.13
        aliases:
          - flask5997_compose
    depends_on:
      - redis


  worker:
    restart: always
    image: pets_celery:1.0 #celery-redis_citc:06.2_ch_term
    depends_on:
      - redis
    secrets:
      - maria_file
      - hadoop_file
      - ahcitcww
    environment:
      - REDIS_HOST:redis_dir
      - REDIS_PORT:6379
      - REDIS_PWD:"citcw200" 
      - MYSQL_PASSWORD_FILE:/run/secrets/db_dba_password 
      #- REDIS_PWD:/run/secrets/my-pw2  

    volumes:
      - ./APP__:/app/app/devp
      - ./config.py:/app/config.py
      - /etc/localtime:/etc/localtime:ro
      - ../hadoop/dataConfig:/home/hadoop/proj_/dataConfig
      - ../hadoop/dataMac:/home/hadoop/proj_/dataMac
      - ../hadoop/data_check:/home/hadoop/proj_/data_check
      - ../hadoop/data_error:/home/hadoop/proj_/data_error


    networks:
      mynetwork:
        ipv4_address: 172.28.1.4
        aliases: 
          - celery_compose    
  
  redis_dir:
    #restart: always
    image: pets_redis:1.0 #pets_reids_alpine:1.0 #redis:4.0.1-alpine02
    ports:
     - "4320:22"
    volumes:
      - /etc/localtime:/etc/localtime:ro  
    
    environment:
      - PWD123:/run/secrets/my-pw2

    entrypoint: redis-server
    command:   --requirepass 'citcw200' --port 6387

    networks:
      mynetwork:
        ipv4_address: 172.28.1.2
        aliases: 
          - redis  
  deidweb:
    restart: always
    image: pets_dir_deidweb:2.7 #pets_dir_deidweb_http:1.0 #pets_dir_deidweb:1.9 #pets_k_deidweb:1.1 pets_dir_deidweb_http:1.0 #
    secrets:
      - ahcitcww 
      - maria_file 
    ports:
     - "11050:11050"
    environment:
      - ASPNETCORE_ENVIRONMENT=Development
      - ASPNETCORE_URLS=http://+:11050;
        #- ASPNETCORE_Kestrel__Certificates__Default__Password=citcw200
        #- ASPNETCORE_Kestrel__Certificates__Default__Path=./server.pfx
    volumes:
      - ./deidweb_log:/app/bin/Debug/netcoreapp3.1/App_Log/servicelog
      - ./appsettings.json:/app/appsettings.json
      - ../hadoop/dataMac/input:/app/wwwroot/uploads/
         # for online debug
         #- ./infra:/app/Infrastructure
         #- ./ctls:/app/Controllers
         #- ./mod:/app/Models
         #- ./views:/app/Views
      - ./server.pfx:/app/server.pfx
      - ./WebAPI.xml:/app/WebAPI.xml
      - ./WebAPI.xml:/app/bin/Debug/netcoreapp3.1/WebAPI.xml
      - ./WebAPI.xml:/app/bin/Release/netcoreapp3.1/WebAPI.xml

    networks:
      mynetwork:
          ipv4_address: 172.28.1.10
          aliases: 
            - deidweb_compose
secrets:
  maria_file:
    external: true
  hadoop_file:
    external: true
  ahcitcww:
    external: true

networks:
  mynetwork:
    external:
      name: hadoopnet_web_overlay
